packages/ember-metal/lib/property_set.js

```diff
@@ -98,10 +97,9 @@ function setPath(root, path, value, tolerant) {
   assert('Property set failed: You passed an empty path', keyName.trim().length > 0)
 
   let newPath = parts.join('.');
-
   let newRoot = getPath(root, newPath);
 
-  if (newRoot) {
+  if (newRoot !== null && newRoot !== undefined) {
```

Comment:
Should this check for `typeof newRoot === “function” || (typeof newRoot === ‘object’ && newRoot !== null)`? 

I don’t think we can call set with other types of objects as the first arg can we?

Reply:
those cases are handled by `set` itself

Reply:
also confirms what is in docs :P https://www.emberjs.com/api/ember/2.15/namespaces/Ember/methods/trySet?anchor=trySet

Reply:
maybe `trySet` could be deprecated because it is using `set` anyway with last parameter as `true` ?

Reply:
or it should be renamed to `trySetPath` because it only works with path :), `trySet(undefined, 'path', 'val')` will still blowup in production

Reply:
added `isDestroyed` check too